client/my-sites/plan-features-2023-grid/index.tsx

```diff
@@ -913,161 +956,179 @@ const ConnectedPlanFeatures2023Grid = connect(
 			currentSitePlanSlug,
 			selectedFeature,
 			intent,
+			highlightCurrentPlan,
 		} = ownProps;
 		const canUserPurchasePlan =
 			! isCurrentPlanPaid( state, siteId ) || isCurrentUserCurrentPlanOwner( state, siteId );
 		const purchaseId = getCurrentPlanPurchaseId( state, siteId );
 		const selectedSiteSlug = getSiteSlug( state, siteId );
+		let currentSitePlanProperties = null;
+
+		const planProperties: PlanProperties[] = plans.reduce(
```

Comment:
This entire change is essentially changing the `.map` to `.reduce`.

An alternative to this was two loops:
```js
filteredPlanProperties = planProperties.filter( plan => !( isCurrentPlan || getPlanClass( plan ) === getPlanClass( currentSitePlanSlug ) );
currentSitePlanProperties = planProperties.find( plan => ( isCurrentPlan || getPlanClass( plan ) === getPlanClass( currentSitePlanSlug ) );
```

If we agree that the above implementation is easier to read, I'll be happy to change this.

Reply:
Cool. My concern is it might create a bunch of pointless conflicts in the other PR for a minor change like this. Maybe just do the two loops implementation if it will avoid conflicts. I'd appreciate anything to help prioritize https://github.com/Automattic/wp-calypso/pull/78731. This whole lot will also get refactored soon (migrate out of plans-features-2023-grid).

Reply:
Thanks @aneeshd16!  https://github.com/Automattic/wp-calypso/pull/78731 is ready. It's mostly tuning types and removing redundant props from typing consuming end.

I'll unblock this PR - If any breakage in https://github.com/Automattic/wp-calypso/pull/78731 then we should proceed as normal with any other work. I'll get back to reviewing your PR tomorrow if none catches me to it. 🙌